File: solutions/797/797-kir3i.cpp
class Solution {
public:
	vector<int> visitLog;
	vector<vector<int>> ans;
	int target;

	vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
		target = graph.size() - 1;
		dfs(graph, 0);
		return ans;
	}

	void dfs(vector<vector<int>> &graph, int curr) {
		visitLog.push_back(curr);
		if (curr == target)
			ans.push_back(vector<int>(visitLog));
		else
			for (int &nxt : graph[curr])
				dfs(graph, nxt);

		visitLog.pop_back();
	}
};

File: solutions/797/797-kir3i.py
class Solution:
    def allPathsSourceTarget(self, graph):
        self.target = len(graph) - 1
        self.ans = []
        self.dfs(graph, 0)

        return self.ans

    def dfs(self, graph, curr, visitLog=[]):
        if curr == self.target:
            self.ans.append(visitLog+[curr])
            return

        for next in graph[curr]:
            print(f'{curr} -> {next}')
            print(visitLog)
            self.dfs(graph, next, visitLog+[curr])
